### ISSUE TITLE
benchmarks for Data.Set.powerSet?

### ISSUE THREAD

**Issue by jwaldmann** [haskell/containers#896]
`Data.Set.powerSet` has a `testProperty` in containers-tests/tests/set-properties.hs, but no benchmarks. Do we want some?

This is related to #890 but I'm making a separate issue because that thread is already quite long, and mainly on implementation details.

Candidates (I can make a PR)
* for construction:
```
whnf S.powerSet (S.fromList[1..w])
```
* for lookup:
```
whnf (\ s -> all (flip S.member s) s) (S.powerSet (S.fromList [1..w]))
```

What should `w` be? For `w = 20`, I get 
```
  powerSet:                  OK (7.22s)
    471  ms ±  32 ms
  member.powerSet:           OK (18.77s)
    2.561 s ± 106 ms
```
that's a lot (w.r.t. to the other benchmarks)

NB: with my suggested implementation (d516927c8b7f5d5803e0cd4a7c335c01b4dbab70), these numbers become (but that really belongs in the other issue)

```
  powerSet:                  OK (0.79s)
    260  ms ± 2.9 ms
  member.powerSet:           OK (15.90s)
    2.216 s ± 1.9 ms
```

**Comment by treeowl** [comment]
The two benchmarks should have different sizes. Make the construction one a bit larger. For the lookup one, I'd like to see an even and and odd size, both a bit smaller.

**Comment by treeowl** [comment]
Closed in #897 .
